### PR TITLE
Feature/support branch prefix

### PR DESCRIPTION
**Pull request by leoxiao2012**
Support custom branch prefixes (e.g. feature/, bugfix/, hotfix/) in auto-generated git branches. The agent chooses the prefix based on context and passes --prefix to the script. Number extraction now handles prefixed branch names correctly.

## Description

  Add --prefix (Bash) / -Prefix (PowerShell) option to create-new-feature scripts, enabling custom branch name prefixes such as feature, bugfix, or hotfix.

  Motivation: Auto-generated branch names have a fixed format like 001-user-auth, which doesn't fit workflows like GitFlow that require category prefixes. Since the prefix varies by context (feature/bugfix/hotfix), it is not suitable as a fixed
  config — the agent decides the appropriate prefix based on the nature of the work.

  Changes:

  - Add --prefix argument to 4 scripts (Bash × 2 + PowerShell × 2)
  - Auto-append /, producing deterministic {prefix}/{number}-{short-name} format (e.g., --prefix "feature" → feature/001-auth)
  - Fix _extract_highest_number / Get-HighestNumberFromNames to correctly extract numbers from prefixed branch names (e.g., feature/003-name), preventing duplicate numbering
  - Update branch name construction and 244-byte truncation logic to account for prefix length
  - Update speckit.git.feature.md command instructions to guide agents on prefix selection
  - Update extensions/git/README.md with --prefix usage documentation

  Backward compatible: omitting --prefix produces identical behavior to before.

## Testing

  - bash create-new-feature.sh --dry-run --json --prefix "feature" --short-name "auth" "test" → {"BRANCH_NAME":"feature/001-auth","FEATURE_NUM":"001","DRY_RUN":true}
  - Without prefix — backward compatible → {"BRANCH_NAME":"001-auth","FEATURE_NUM":"001","DRY_RUN":true}
  - --help output includes new option

## AI Disclosure

- [ ] I **did not** use AI assistance for this contribution
- [x] I **did** use AI assistance (describe below)

Code generated by Claude Code (Opus 4.6), reviewed and verified manually.

### CODE REVIEW

**Review by mnriem — CHANGES_REQUESTED** [review]
Please address Copilot feedback

**Comment by mnriem** [comment]
Please address Copilot feedback. If not applicable, please explain why

**Comment by leoxiao2012** [comment]
Add input validation for --prefix/-Prefix across all 4 scripts (Bash,
PowerShell, extension variants):

- Reject non-ASCII characters and special characters
- Require prefix to start with a letter or digit (^[a-z0-9][-a-z0-9]*$)
- Enforce max length of 16 characters via extracted MAX_PREFIX_LEN constant
- Reject slash-only values (e.g. "/", "//") and embedded slashes
- Add comprehensive test coverage for all edge cases

File: extensions/git/scripts/bash/create-new-feature.sh [hunk 1]
```diff
@@ -59,8 +61,21 @@ while [ $i -le $# ]; do
...
+            if [ $((i + 1)) -gt $# ]; then
+                echo 'Error: --prefix requires a value' >&2
+                exit 1
+            fi
+            i=$((i + 1))
+            next_arg="${!i}"
+            if [[ "$next_arg" == --* ]]; then
+                echo 'Error: --prefix requires a value' >&2
+                exit 1
+            fi
```
[the remark this answers is not in the document]
Comment (leoxiao2012) [reply extensions/git/scripts/bash/create-new-feature.sh:74]: Not an issue — working as intended.                                                                                                                                                              
                                                                                                                                                                                                   
  The --prefix parser follows the same pattern as --short-name and --number: it guards against a missing next argument or a ---prefixed value. Empty strings are a separate concern and are already
   handled downstream at lines 108-113:                                                                                                                                                            
```shell                                                                                         
  if [ -n "$BRANCH_PREFIX" ]; then                                                                                                                                                                 
      BRANCH_PREFIX=$(echo "$BRANCH_PREFIX" | sed 's/^[[:space:]]*//;s/[[:space:]]*$//')                                                                                                           
      if [ -z "$BRANCH_PREFIX" ]; then                                                                                                                                                             
          echo 'Error: --prefix cannot be empty or whitespace' >&2                                                                                                                                 
          exit 1                                                                                                                                                                                   
      fi                                                                                                                                                                                           
      ...                                                                                                                                                                                          
  fi  
```

File: extensions/git/scripts/powershell/create-new-feature.ps1 [hunk 2]
```diff
@@ -37,12 +41,39 @@ if ($Help) {
...
 
 $featureDesc = ($FeatureDescription -join ' ').Trim()
 
+# Validate and normalize branch prefix
+if ($Prefix) {
+    $Prefix = $Prefix.Trim()
+    if ([string]::IsNullOrWhiteSpace($Prefix)) {
+        Write-Error "Error: -Prefix cannot be empty or whitespace"
+        exit 1
+    }
```
[the remark this answers is not in the document]
Comment (leoxiao2012) [reply extensions/git/scripts/powershell/create-new-feature.ps1:56]: Not an issue — working as intended. Explained above.

File: scripts/powershell/create-new-feature.ps1 [hunk 3]
```diff
@@ -37,12 +41,39 @@ if ($Help) {
...
 
 $featureDesc = ($FeatureDescription -join ' ').Trim()
 
+# Validate and normalize branch prefix
+if ($Prefix) {
+    $Prefix = $Prefix.Trim()
+    if ([string]::IsNullOrWhiteSpace($Prefix)) {
+        Write-Error "Error: -Prefix cannot be empty or whitespace"
+        exit 1
+    }
```
[the remark this answers is not in the document]
Comment (leoxiao2012) [reply scripts/powershell/create-new-feature.ps1:56]: Not an issue — working as intended. Explained above.

File: scripts/bash/create-new-feature.sh [hunk 4]
```diff
@@ -52,8 +54,22 @@ while [ $i -le $# ]; do
         --timestamp)
             USE_TIMESTAMP=true
             ;;
+        --prefix)
+            if [ $((i + 1)) -gt $# ]; then
+                echo 'Error: --prefix requires a value' >&2
+                exit 1
+            fi
```
[the remark this answers is not in the document]
Comment (leoxiao2012) [reply scripts/bash/create-new-feature.sh:61]: Not an issue — working as intended.

The --prefix parser follows the same pattern as --short-name and --number: it guards against a missing next argument or a ---prefixed value.

**Comment by mnriem** [comment]
Please address Copilot feedback

**Review by mnriem — DISMISSED** [review]

**Comment by mnriem** [comment]
Please address test & lint errors

**Review by mnriem — CHANGES_REQUESTED** [review]
Please address Copilot feedback

**Comment by mnriem** [comment]
Thanks for the work here, @leoxiao2012. After review, this PR has been superseded by functionality already merged into `main`, so I'm going to close it.

Custom / GitFlow-style branch prefixes are now supported through the git extension **config** rather than an agent-passed flag:

- **#2202** (merged) added GitFlow single-segment prefix handling so refs like `feat/001-my-feature` are accepted by validation and resolve their `specs/` path correctly.
- **#3293** (merged) added a `branch_prefix` shorthand plus a full `branch_template` (with `{author}`, `{app}`, `{number}`, `{slug}` tokens), with namespace-aware numbering and validation.

Today you can get `feature/001-auth` by setting the following in `extensions/git/git-config.yml`:

```yaml
branch_prefix: "feature"
# or, for full control:
branch_template: "{number}-{slug}"
```

On top of that, the scripts this PR modifies have since been renamed (`create-new-feature.sh` → `create-new-feature-branch.sh`) and then ported to Python (**#3400**), so the diff no longer applies (the PR is currently in a conflicting state).

Config-based prefixes were preferred over a per-invocation flag because the prefix is generally stable per project/workflow rather than something the agent should decide each time. Closing as superseded — but thank you for pushing on this; it's a real use case that's now covered.

_Posted on behalf of @mnriem by GitHub Copilot (model: Claude Opus 4.8)._
